### PR TITLE
use const ref in boost::fusion::vector, fix #511

### LINKED ISSUE #511: Const correctness in boost::fusion::vector for ArgsType & Boost versions
Hi,

If we consider this simple exemple:
```
#include <boost/fusion/container/vector.hpp>

struct Base
{
    typedef boost::fusion::vector<const unsigned long, int> ArgsType;
    ArgsType args;

    Base(ArgsType args): args(args) {};
};

int main()
{
    Base::ArgsType args(3, 4);
    Base instance(args);
    return 0;
}
```

It compiles on Boost 1.64, 1.65, doesn't n 1.66 & 1.67, and compiles again in 1.68, as you can see here: https://godbolt.org/z/aHrEBe

When it doesn't, the issue is with the `const` keyword:
```
/opt/compiler-explorer/libs/boost_1_67_0/boost/fusion/container/vector/detail/deref_impl.hpp:47:49: error: binding reference of type 'boost::fusion::extension::deref_impl<boost::fusion::vector_iterator_tag>::apply<boost::fusion::vector_iterator<boost::fusion::vector<const long unsigned int, int>, 0> >::type' {aka 'long unsigned int&'} to 'const long unsigned int' discards qualifiers
```

I will dig a bit, but I think this is an issue in Boost, so I suggest to simply avoid those versions.

### PR DESCRIPTION
**Pull request by nim65s**
With this, pinocchio & py-pinocchio are building correctly with robotpkg in fedora 28, which uses boost 1.66.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ cache:
     - git pull
     - cd pinocchio
     - make checkout MASTER_REPOSITORY="dir ${CI_PROJECT_DIR}"
+    - export GIT_DIR=${CI_PROJECT_DIR}/.git
```
Comment (jcarpent) [inline .gitlab-ci.yml]: Good catch!
@nim65s Do you know you have to provide this hint?

**Comment by jcarpent** [comment]
@nim65s Any news for this PR?

**Comment by nim65s** [comment]
no, sorry, I am in parental leave until 19/11 :)

**Comment by nim65s** [comment]
Still, these are not the commit that should be in this PR… I will try to find it.

**Comment by nim65s** [comment]
Sorry for the noise, I'll open another PR as soon as my tests will be done

**Comment by jcarpent** [comment]
Ok. No problem.

**Pull request closed without merging**